File: 2026POTD/JANUARY/25-01-2026/1984.rb
# @param {Integer[]} nums
# @param {Integer} k
# @return {Integer}
def minimum_difference(nums, k)
  return 0 if k <= 1
  nums.sort!
  ans = Float::INFINITY
  (0..nums.length - k).each do |i|
    ans = [ans, nums[i + k - 1] - nums[i]].min
  end
  ans
end

File: 2026POTD/JANUARY/25-01-2026/1984.ex
defmodule Solution do
  @spec minimum_difference(nums :: [integer], k :: integer) :: integer
  def minimum_difference(nums, k) do
    if k <= 1 do
      0
    else
      nums = Enum.sort(nums)
      nums
      |> Enum.chunk_every(k, 1, :discard)
      |> Enum.map(fn chunk -> List.last(chunk) - hd(chunk) end)
      |> Enum.min()
    end
  end
end